### PR TITLE
refactor(core): move error to separate module

### PR DESCRIPTION
**Pull request by ozgrakkurt**
Moved error type to separate file inside core crate

### CODE REVIEW

**Review by aasseman — CHANGES_REQUESTED** [review]
Please fix the failed CI checks by reviewing:
- https://github.com/semiotic-ai/timeline-aggregation-protocol/blob/main/CONTRIBUTING.md#commit-messages
- https://github.com/semiotic-ai/timeline-aggregation-protocol/blob/main/CONTRIBUTING.md#i-want-to-contribute

File: tap_core/src/lib.rs [hunk 1]
```diff
@@ -8,49 +8,16 @@
...
 use std::time::{SystemTime, UNIX_EPOCH};
 
-use ethereum_types::Address;
-use ethers::{signers::WalletError, types::SignatureError};
-use receipt_aggregate_voucher::ReceiptAggregateVoucher;
 use thiserror::Error;
 
 pub mod adapters;
 pub mod eip_712_signed_message;
+mod error;
```
Comment (ColePBryan) [inline tap_core/src/lib.rs:15]: I think this needs to be made pub since the user of the lib would need to be able to handle the errors.

**Review by ColePBryan — CHANGES_REQUESTED** [review]
This refactor is a good call, definitely makes it more readable.

File: tap_core/src/lib.rs (hunk 1 above)
Comment (ozgrakkurt) [reply tap_core/src/lib.rs:15]: it is re-exported from the root of the crate

**Review by ColePBryan — APPROVED** [review]

**Review by aasseman — APPROVED** [review]

**Pull request merged**